src/video/sdl_v.cpp

```diff
@@ -346,39 +330,42 @@ bool VideoDriver_SDL::CreateMainSurface(uint w, uint h)
 	 * surface, for example). */
 	_requested_hwpalette = want_hwpalette;
 
-	/* DO NOT CHANGE TO HWSURFACE, IT DOES NOT WORK */
-	newscreen = SDL_SetVideoMode(w, h, bpp, SDL_SWSURFACE | (want_hwpalette ? SDL_HWPALETTE : 0) | (_fullscreen ? SDL_FULLSCREEN : SDL_RESIZABLE));
-	if (newscreen == NULL) {
+	seprintf(caption, lastof(caption), "OpenTTD %s", _openttd_revision);
+
+	if (_sdl_window == NULL) {
+		Uint32 flags = SDL_WINDOW_SHOWN;
+
+		if (_fullscreen) {
+			flags |= SDL_WINDOW_FULLSCREEN_DESKTOP;
+		} else {
+			flags |= SDL_WINDOW_RESIZABLE;
+		}
+
+		_sdl_window = SDL_CreateWindow(
+			caption,
+			SDL_WINDOWPOS_UNDEFINED,
+			SDL_WINDOWPOS_UNDEFINED,
+			w, h,
+			flags);
+	}
+
+	if (_sdl_window == NULL) {
 		DEBUG(driver, 0, "SDL: Couldn't allocate a window to draw on");
 		return false;
 	}
+
+	SDL_SetWindowSize(_sdl_window, w, h);
```

Comment:
Is this call necessary? If I remove it, I can resize the game window properly, and as far as I can see nothing breaks.

Reply:
This seems to be necessary for changing the window size with the "screen resolution" dropdown in the game options. When I remove this, I can still resize the window by dragging the window edges via the window manager, but then the "screen resolution" dropdown doesn't change the window size.

Reply:
Yup, I realised. Check out the PR -- https://github.com/nikolas/OpenTTD/pull/1